pypdf/filters.py

```diff
@@ -695,19 +718,36 @@ def _xobj_to_image(x_object_obj: Dict[str, Any]) -> Tuple[Optional[str], bytes]:
                 mode_map = {
                     "/DeviceGray": "L",
                     "/DeviceRGB": "RGB",
-                    "/DeviceCMYK": "RGBA",
+                    "/DeviceCMYK": "CMYK",  # used to be "RGBA" but this is seems not in accordance withFlateEncode Spec
```

Comment:
I agree with this change - I don't think we need the comment here

```suggestion
                    "/DeviceCMYK": "CMYK",
```

Reply:
applied